### DISCUSSION TITLE
Is posible to run test in  "--dry-run" mode ?

### DISCUSSION THREAD

**Discussion by despotbg** [percy/percy-puppeteer#344] [Q&A] [answered]
Is possible to run test in  "--dry-run" mode? This is handy while making tests 

thx

**Comment by Robdel12** [comment]
Hey @despotbg! I'm not quite sure I follow. `--dry-run` will make it so DOM snapshots are captured and run through asset discovery, but not uploaded to Percy for screenshots (so you're not using up snapshots debugging). Screenshots aren't captured locally, those are captured in Percy's infra (and what you pay for)

**Reply by despotbg** [reply]
Hey @Robdel12 
Sorry for the short question, I will try to explain what I want 

See here, there is the option for dry-run when using cli-snapshot  https://docs.percy.io/docs/cli-snapshot#page-listing

But if I use Puppeteer SDK, I will run the test with `percy exec`. Here https://github.com/percy/cli/tree/master/packages/cli-exec#percy-exec, there is no option to make test "dry"?

I run my test with this line `npx percy exec -- node ./percy/snapshotsNEW.js`

**Comment by wwilsman** [comment, marked as answer]
Hey @despotbg! What are you looking for out of a `--dry-run` feature?

Are you looking to skip Percy snapshots altogether? If so, you can simply run your tests without `percy exec`.

Are you looking to run asset discovery for debugging, but not upload snapshots? If so, this would be the `--debug` flag in the latest version of the `@percy/cli`.

Or are you looking for something different? For example, some commands have a `--dry-run` flag that simply prints the names of snapshots. This isn't implemented in our `exec` command due to the snapshot names being printed in the middle of test output for most SDKs and you would need to parse through logs to find snapshot names.

If that last option is what you're looking for, but you don't have the issue of test output getting in the way, maybe you could stub the `percySnapshot` function to `console.log` inside your Puppeteer script?

**Reply by despotbg** [reply]
Hi @wwilsman 
🤦‍♂️ This is exactly what I need _"run your tests without percy exec"_   
Thank you!
